openedx/core/djangoapps/user_api/preferences/tests/test_views.py

```diff
@@ -0,0 +1,484 @@
+        self.send_patch(self.client, {
+            "string_pref": "updated_value",
+            "new_pref": "new_value",
+            "extra_pref": None,
+        })
+
+        # Verify that GET returns the updated preferences
+        response = self.send_get(self.client)
+        expected_preferences = {
+            "dict_pref": "{'int_key': 10}",
+            "string_pref": "updated_value",
+            "new_pref": "new_value",
+        }
+        self.assertEqual(expected_preferences, response.data)
+
+    def test_update_preferences_bad_data(self):
+        """
+        Test that a client (logged in) receives appropriate errors for a bad update.
+        """
+        # Create some test preferences values.
+        UserPreference.set_preference(self.user, "dict_pref", {"int_key": 10})
+        UserPreference.set_preference(self.user, "string_pref", "value")
+        UserPreference.set_preference(self.user, "extra_pref", "extra_value")
+
+        # Send the patch request
+        self.client.login(username=self.user.username, password=self.test_password)
+        response = self.send_patch(
+            self.client,
+            {
+                "string_pref": "updated_value",
+                TOO_LONG_PREFERENCE_KEY: "new_value",
+                "new_pref": "new_value",
+            },
+            expected_status=400
+        )
+        expected_message = u"The user preference has the following errors: " \
+                           "{'key': [u'Ensure this value has at most 255 characters (it has 256).']}"
+        self.assertEquals(
+            response.data,
+            {
+                "field_errors": {
+                    TOO_LONG_PREFERENCE_KEY: {
+                        "developer_message": expected_message
+                    }
+                },
+            }
+        )
+
+        # Verify that GET returns the original preferences
+        response = self.send_get(self.client)
+        expected_preferences = {
+            u"dict_pref": u"{'int_key': 10}",
+            u"string_pref": u"value",
+            u"extra_pref": u"extra_value",
+        }
+        self.assertEqual(expected_preferences, response.data)
+
+    def test_update_preferences_bad_request(self):
+        """
+        Test that a client (logged in) receives appropriate errors for a bad request.
+        """
+        self.client.login(username=self.user.username, password=self.test_password)
+
+        # Verify a non-dict request
+        response = self.send_patch(self.client, "non_dict_request", expected_status=400)
+        self.assertEqual(
+            response.data,
+            {
+                "developer_message": u"No data provided for user preference update",
+                "user_message": u"No data provided for user preference update"
+            }
+        )
+
+        # Verify an empty dict request
+        response = self.send_patch(self.client, {}, expected_status=400)
```

Comment:
I can be convinced that it doesn't need to throw an error. I just thought it was likely indicative of a client error. We should be consistent across all of our PATCH implementations which argues for not doing this since none of the others are.